tfe/resource_tfe_workspace_test.go

```diff
@@ -2249,6 +2297,22 @@ resource "tfe_workspace" "foobar" {
 }`, organization, organization)
 }
 
+// while testing the flow of unsetting execution mode as in TestAccTFEWorkspace_unsetExecutionMode
+// the resource "tfe_agent_pool" has been kept in both configs(testAccTFEWorkspace_executionModeAgent & testAccTFEWorkspace_executionModeNull)
+// this prevents an attempt to destroy the agent pool before dissasociating it from the workspace
+func testAccTFEWorkspace_executionModeNull(organization string) string {
+	return fmt.Sprintf(`
+resource "tfe_agent_pool" "foobar" {
```

Comment:
The test was failing during the agent mode unset flow because the first config had the `resource "tfe_agent_pool"` and the second config did not.The diff causes an attempt to delete the agent pool while applying the second config.
~ Approach 1 is to persist the `resource "tfe_agent_pool"` in both configs so the only diff action is unsetting the workspace execution mode.
~Approach 2 is to create agent pool external to the config and then use data source to grab the agent pool in the config. This would required using the go client interface.

I have gone with approach 1. Let me know your thoughts...

Reply:
Much, much nicer than approach 2 :+1: